Build failure: libapparmor (`armv7l` cross)

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

```
nix build .#pkgsCross.armv7l-hf-multiplatform.libapparmor
```

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/5s1xk3i1ij7rwnnn97chlzlyxaiypi09-source
source root is source/libraries/libapparmor
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: autoreconfPhase
@nix { "action": "setPhase", "phase": "autoreconfPhase" }
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.ac,
libtoolize: and rerunning libtoolize and aclocal.
libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force 
autoreconf: running: /nix/store/3m2741qdz8rgbl2mkx8rciwgav3gc7b4-autoconf-2.72/bin/autoconf --force
configure.ac:8: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.
./lib/autoconf/general.m4:2434: AC_DIAGNOSE is expanded from...
aclocal.m4:9950: AM_INIT_AUTOMAKE is expanded from...
configure.ac:8: the top level
configure.ac:10: warning: AC_PROG_LEX without either yywrap or noyywrap is obsolete
./lib/autoconf/programs.m4:743: _AC_PROG_LEX is expanded from...
./lib/autoconf/programs.m4:736: AC_PROG_LEX is expanded from...
aclocal.m4:10189: AM_PROG_LEX is expanded from...
configure.ac:10: the top level
configure.ac:48: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:48: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
m4/ac_python_devel.m4:1: AC_PYTHON_DEVEL is expanded from...
configure.ac:48: the top level
configure.ac:81: warning: The macro 'AC_HEADER_STDC' is obsolete.
configure.ac:81: You should run autoupdate.
./lib/autoconf/headers.m4:663: AC_HEADER_STDC is expanded from...
configure.ac:81: the top level
configure.ac:88: warning: The macro 'AM_PROG_LIBTOOL' is obsolete.
configure.ac:88: You should run autoupdate.
aclocal.m4:399: AM_PROG_LIBTOOL is expanded from...
configure.ac:88: the top level
configure.ac:90: warning: The macro 'AC_PROG_CC_C99' is obsolete.
configure.ac:90: You should run autoupdate.
./lib/autoconf/c.m4:1662: AC_PROG_CC_C99 is expanded from...
configure.ac:90: the top level
configure.ac:106: warning: AC_OUTPUT should be used without arguments.
configure.ac:106: You should run autoupdate.
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:8: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.  For more info, see:
configure.ac:8: https://www.gnu.org/software/automake/manual/automake.html#Modernize-AM_005fINIT_005fAUTOMAKE-invocation
configure.ac:10: installing './compile'
configure.ac:88: installing './config.guess'
configure.ac:88: installing './config.sub'
configure.ac:8: installing './install-sh'
configure.ac:8: installing './missing'
doc/Makefile.am:10: warning: subst .2,.pod,$(man_MANS: non-POSIX variable name
doc/Makefile.am:10: (probably a GNU make extension)
doc/Makefile.am:10: warning: subst .3,.pod,$(man_MANS: non-POSIX variable name
doc/Makefile.am:10: (probably a GNU make extension)
doc/Makefile.am:17: warning: '%'-style pattern rules are a GNU make extension
doc/Makefile.am:26: warning: '%'-style pattern rules are a GNU make extension
src/Makefile.am:69: warning: '%'-style pattern rules are a GNU make extension
src/Makefile.am:1: warning: 'INCLUDES' is the old name for 'AM_CPPFLAGS' (or '*_CPPFLAGS')
src/Makefile.am: installing './depcomp'
configure.ac: installing './ylwrap'
parallel-tests: installing './test-driver'
testsuite/Makefile.am:8: warning: 'INCLUDES' is the old name for 'AM_CPPFLAGS' (or '*_CPPFLAGS')
autoreconf: './install-sh' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing libtool script ./ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/xg75pc4yyfd5n2fimhb98ps910q5lm5n-bash-5.2p37/bin/sh"
configure flags: --disable-dependency-tracking --prefix=/nix/store/f16r2975wlbkc9xbfnazgridfj505ykr-libapparmor-armv7l-unknown-linux-gnueabihf-4.1.0 --without-perl --with-python --build=x86_64-unknown-linux-gnu --host=armv7l-unknown-linux-gnueabihf
checking for a BSD-compatible install... /nix/store/cg09nslw3w6afyynjw484b86d47ic1cb-coreutils-9.7/bin/install -c
checking whether build environment is sane... yes
checking for armv7l-unknown-linux-gnueabihf-strip... armv7l-unknown-linux-gnueabihf-strip
checking for a race-free mkdir -p... /nix/store/cg09nslw3w6afyynjw484b86d47ic1cb-coreutils-9.7/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports the include directive... yes (GNU style)
checking for armv7l-unknown-linux-gnueabihf-gcc... armv7l-unknown-linux-gnueabihf-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether armv7l-unknown-linux-gnueabihf-gcc accepts -g... yes
checking for armv7l-unknown-linux-gnueabihf-gcc option to enable C11 features... none needed
checking whether armv7l-unknown-linux-gnueabihf-gcc understands -c and -o together... yes
checking dependency style of armv7l-unknown-linux-gnueabihf-gcc... none
checking for flex... flex
checking for lex output file root... lex.yy
checking for lex library... none needed
checking for library containing yywrap... no
checking whether yytext is a pointer... yes
checking for bison... bison -y
checking for a sed that does not truncate output... /nix/store/1h8gf327cgid0jgjygrj31amp63mn7a7-gnused-4.9/bin/sed
checking pkg-config is at least version 0.9.0... yes
checking for swig... /nix/store/i68vky1ll3qs24qgpy3cl8q3nrprnj8i-swig-4.3.0/bin/swig
checking whether the libapparmor debug output should be enabled... no
checking whether the libapparmor man pages should be generated... yes
checking for podchecker... podchecker
checking for pod2man... pod2man
checking whether python bindings are enabled... yes
checking for python3... /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3
checking for python... (cached) /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3
checking for armv7l-unknown-linux-gnueabihf-python3-config... no
checking for python3-config... /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config
configure: WARNING: using cross tools not prefixed with host triplet
checking for a version of Python >= '2.1.0'... yes
checking for the setuptools Python package... yes
checking for Python include path... /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config is /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config
-I/nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/include/python3.12 -I/nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/include/python3.12
checking for Python library path... /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config is /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config
 -L/nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/lib  -ldl -L/nix/store/8q0sgpqv4km05yfxw00silpvbizjbqdj-libxcrypt-4.4.38/lib -lm 
checking for Python site-packages path... /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/lib/python3.12/site-packages
checking python extra libraries... /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config is /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config
-lpython3.12 -ldl -L/nix/store/8q0sgpqv4km05yfxw00silpvbizjbqdj-libxcrypt-4.4.38/lib -lm 
checking python extra linking flags... /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config is /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/bin/python3-config
 -L/nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/lib -lpython3.12 -ldl -L/nix/store/8q0sgpqv4km05yfxw00silpvbizjbqdj-libxcrypt-4.4.38/lib -lm 
checking consistency of all components of python development environment... no
configure: error: 
  Could not link test program to Python. Maybe the main Python library has been
  installed in some non-standard library path. If so, pass it to configure,
  via the LDFLAGS environment variable.
  Example: ./configure LDFLAGS="-L/usr/non-standard-path/python/lib"
  ============================================================================
   ERROR!
   You probably have to install the development version of the Python package
   for your distribution.  The exact name of this package varies among them.
  ============================================================================
```

### Additional context

Bisected the build failure to eba90bba663352aec63632d2547c07e19414923b (https://github.com/NixOS/nixpkgs/pull/400430).

The issue does _not_ seem to affect `aarch64-multiplatform` cross.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.83, NixOS, 24.11 (Vicuna), 24.11.20250320.7105ae3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`

### Notify maintainers

@ju1m @thoughtpolice @LordGrimmauld 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

I have now been pointed to the configure log, which complains:
```
configure:6059: armv7l-unknown-linux-gnueabihf-gcc -o conftest -g -O2  -I/nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/include/python3.12 -I/nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/include/python3.12  conftest.c  -lpython3.12 -ldl -L/nix/store/8q0sgpqv4km05yfxw00silpvbizjbqdj-libxcrypt-4.4.38/lib -lm   -L/nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/lib  -ldl -L/nix/store/8q0sgpqv4km05yfxw00silpvbizjbqdj-libxcrypt-4.4.38/lib -lm  >&5
In file included from /nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/include/python3.12/Python.h:38,
                 from conftest.c:13:
/nix/store/kjvgj2n3yn70hmjifg6y0bk9m4rf7jba-python3-3.12.10/include/python3.12/pyport.h:586:2: error: #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
  586 | #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
      |  ^~~~~
configure:6059: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PACKAGE "libapparmor1"
| #define VERSION "4.1.0"
| #define YYTEXT_POINTER 1
| /* end confdefs.h.  */
| 
|                 #include <Python.h>
| 
| int
| main (void)
| {
| 
|                 Py_Initialize();
| 
|   ;
|   return 0;
| }
configure:6069: result: no
configure:6073: error: 
  Could not link test program to Python. Maybe the main Python library has been
  installed in some non-standard library path. If so, pass it to configure,
  via the LDFLAGS environment variable.
  Example: ./configure LDFLAGS="-L/usr/non-standard-path/python/lib"
  ============================================================================
   ERROR!
   You probably have to install the development version of the Python package
   for your distribution.  The exact name of this package varies among them.
  ============================================================================
```

`LONG_BIT definition appears wrong for platform` is a bit of a cryptic error, but it seems one can't cross-compile with python with a different bit depth between host and build.

--- a/pkgs/by-name/li/libapparmor/package.nix
+++ b/pkgs/by-name/li/libapparmor/package.nix
@@ -12,7 +12,11 @@
     stdenv.hostPlatform == stdenv.buildPlatform && lib.meta.availableOn stdenv.hostPlatform perl,
   perl,
   withPython ?
-    !stdenv.hostPlatform.isStatic && lib.meta.availableOn stdenv.hostPlatform python3Packages.python,
+    # static can't load python libraries
+    !stdenv.hostPlatform.isStatic
+    && lib.meta.availableOn stdenv.hostPlatform python3Packages.python
+    # m4 python include script fails if cpu bit depth is different across machines
+    && stdenv.hostPlatform.parsed.cpu.bits == stdenv.buildPlatform.parsed.cpu.bits,
   python3Packages,
   swig,
   ncurses,

